text/stacks/0007-squash-stack.md

```diff
@@ -0,0 +1,45 @@
+# Squash Stack Images
+
+## Summary
+
+The build and run images delivered as a stack for our users should be squashed
+into a single layer to reduce the total number of layers in the builder and
+application images and remove duplicated files that appear in multiple layers.
+
+## Motivation
+
+Stack images act as the base image for applications and our own builder images.
+Their internal layer implementations should not be important to end users.
+Recently, we have been fielding a lot of bug reports wherein users cannot
+create their application images because the total number of layers for the
+image has exceeded the limit enforced by `containerd`.
+
+Squashing the stack images down into a single layer will create more headroom
+for both the builders and the images they create.
```

Comment:
What do you envision the `builder.toml` to look like when using a single layer? Will the separate `build-image` and `run-image` lines go away in favour of a `stack-image` line? Have you tried this out in a builder to see if it works as expected?

Reply:
Sorry. This is just clumsily worded. There are still 2 images. Its just that each image only has 1 layer.

Reply:
Oh, ok, yea that didn't come through. I was a confused about how that would work

Reply:
I think I should say "Squashing each stack image down into a single layer..."

Is that clearer?

Reply:
Definitely!